fallback to a working nameserver if dns does not work in the chroot env

lausser opened the pull request:
i was running a build in a corporate intranet and found out that a nameserver 1.0.0.1 does not work here. So when
chroot "${SDCARD}" /bin/bash -c "/tmp/install.sh"
was run, all the apt-commands in the install.sh script ended in a timout. So i try to resolve apt.armbian.com first and if that fails, use the resolv.conf from the build environment.

count-doku commented:
Hey,

I like the idea. But why do you 'cd' into /tmp if you are using fully qualified paths anyway? And because you never cd back (to wherever we were before) it might break something later on.

Could you remove the 'cd' in line 338?

lausser commented:
My intention was to have wget write the index.html to a place where it does not interfere with something else. But, as i rm it anyway, the folder is not relevant. So i'll remove the cd.
Another thing came to my mind. The trick with the "outside" resolv.conf might help during the build process, but then it stays in the image. Now if you use the sd-card with this image elsewhere, you have some intranet configuration which might not work. So i thought it will be a good idea, so save the original resolv.conf with the 1.0.0.1 inside and write it back to /etc at the end of the /tmp/install.sh.
What do you think?

count-doku commented:
Regarding cd to tmp: 
Maybe save the old directory, then cd to tmp, after your code skip back to the original directory.

Regarding dns config: 
Definitely restore the original resolv.conf with 1.0.0.1 at the end. It is much more likely that 1.0.0.1 works in a random network than the one on the build host.

lausser commented:
Now go back to the start directory.
Save and restore the original resolv.conf
Cleanup /tmp

zhangn1985 commented:
@lausser what you are doing is wrong, build system provide a varible `NAMESERVER=xxxxxxx` to let you to specify name server.

please add it to your own config file in `userpatch` folder or add it in build command line.

count-doku commented:
@lausser does the `NAMESERVER` parameter satisfy your use-case?

lausser commented:
Just found the time to try....setting the environment variable NAMESERVER does not work. I added 
```
        cat <<-EOF > "${SDCARD}"/tmp/install.sh
        #!/bin/bash
echo NAMESERVER is \$NAMESERVER
echo resolv.conf is
cat /etc/resolv.conf
        [[ "$remote_only" != yes ]] && apt-key add /tmp/buildpkg.key

```in chroot-buildpackages.sh and it still showed 1.0.0.1

Added NAMESERVER=...my-dns-ip to userpatches/lib.config and it worked fine, setting the correct resolv.conf inside the chrooted install-script.

So this pull request can be discarded.
